README.md

```diff
@@ -114,43 +163,276 @@ encoding as JSON. Since Python is a dynamic language, when decoding via
+* A single argument to the child workflow is positional. Multiple arguments are not supported in the type-safe form of
+  start/execute child workflow and must be supplied via the `args` keyword argument.
+* Child workflow options are set as keyword arguments after the arguments. At least `id` must be provided.
+* The `await` of the start does not complete until the workflow has confirmed to be started
+* The result is a child workflow handle which is an `asyncio.Task` and supports basic task features. The handle also has
+  some child info and supports signalling the child workflow
+* An async `workflow.execute_child_workflow()` helper is provided which takes the same arguments as
+  `workflow.start_child_workflow()` and `await`s on the result. This should be used in most cases unless advanced task
+  capabilities are needed.
+
+#### Timers
+
+* A timer is represented by normal `asyncio.sleep()`
+* Timers are also implicitly started on any `asyncio` calls with timeouts (e.g. `asyncio.wait_for`)
+* Timers are Temporal server timers, not local ones, so sub-second resolution rarely has value
+
+#### Conditions
+
+* `workflow.wait_condition` is an async function that doesn't return until a provided callback returns true
+* A `timeout` can optionally be provided which will throw a `asyncio.TimeoutError` if reached (internally backed by
+  `asyncio.wait_for` which uses a timer)
+
+#### Asyncio and Cancellation
+
+Workflows are backed by a custom [asyncio](https://docs.python.org/3/library/asyncio.html) event loop. This means many
+of the common `asyncio` calls work as normal. Some asyncio features are disabled such as:
+
+* Thread related calls such as `to_thread()`, `run_coroutine_threadsafe()`, `loop.run_in_executor()`, etc
+* Calls that alter the event loop such as `loop.close()`, `loop.stop()`, `loop.run_forever()`,
+  `loop.set_task_factory()`, etc
+* Calls that use a specific time such as `loop.call_at()`
+* Calls that use anything external such as networking, subprocesses, disk IO, etc
+
+Cancellation is done the same way as `asyncio`. Specifically, a task can be requested to be cancelled but does not
+necessarily have to respect that cancellation immediately. This also means that `asyncio.shield()` can be used to
+protect against cancellation. The following tasks, when cancelled, perform a Temporal cancellation:
+
+* Activities - when the task executing an activity is cancelled, a cancellation request is sent to the activity
+* Child workflows - when the task starting or executing a child workflow is cancelled, a cancellation request is sent to
+  cancel the child workflow
+* Timers - when the task executing a timer is cancelled (whether started via sleep or timeout), the timer is cancelled
+
+When the workflow itself is requested to cancel, `Task.cancel` is called on the main workflow task. Therefore,
+`asyncio.CancelledError` can be caught in order to handle the cancel gracefully.
```

Comment:
Can't tell from this how to cancel or what a task is from a code POV

Reply:
This is a Pythonism. I don't think it's reasonable to document Python `asyncio` cancellation here. But once real documentation for the Python SDK comes and all of this is deleted, it may make sense to expand on what a task is or how to cancel tasks in Python.

Reply:
👍 cc @rachfop